### PR TITLE
Add search box with suggestions #LMR-179

### PR DESCRIPTION
**Pull request by livthomas**
I have improved the search box to be able to suggest as well as build queries.

### CODE REVIEW

File: src/main/scripts/src/app/core/header/header.component.html [hunk 1]
```diff
@@ -4,16 +4,12 @@
       <span>Lumeer</span>
     </a>
   </div>
-  <!-- TODO search box will be provided by another component -->
   <div class="search">
-    <input class="search-box" placeholder="Type anything you search for..."/>
-    <button class="search-button">
-      <span>Search</span>
-    </button>
+    <search-box *ngIf="workspaceService.isWorkspaceSet()"></search-box>
```
Comment (kubedo8) [inline src/main/scripts/src/app/core/header/header.component.html]: Isn't property `showSearchBox` planned for this situation?

File: src/main/scripts/src/app/core/rest/search.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@Injectable()
+export class SearchService {
+
+  constructor(private http: Http, private workspaceService: WorkspaceService) {
+  }
+
+  public suggest(text: string, type: string): Observable<Suggestion[]> {
+    return this.http
+      .get(`${this.searchPath()}/suggestion?text=${text}&type=${type}`)
+      .map(response => response.json() as Suggestion[]);
```
Comment (kubedo8) [inline src/main/scripts/src/app/core/rest/search.service.ts]: There is better way to add query parameters, but we will use new `HttpClient` instead and change it...

**Comment by kubedo8** [comment]
Can you include some picture to demonstrate new functionality?

File: src/main/scripts/src/app/core/rest/search.service.ts (hunk 2 above)
Comment (livthomas) [inline src/main/scripts/src/app/core/rest/search.service.ts]: Thanks, I haven't even used our `HttpClient` here. I will refactor this using a new Angular client.

File: src/main/scripts/src/app/core/header/header.component.html (hunk 1 above)
Comment (livthomas) [inline src/main/scripts/src/app/core/header/header.component.html]: Good catch!

**Comment by livthomas** [comment]
screencapture-localhost-7000-ui-w-lmr-devel-search-1500465554759

**Comment by livthomas** [comment]
It looks like this now:
screencapture-localhost-7000-ui-w-lmr-devel-search-1500541555099

**Pull request merged**